tests/scripts/moved-redirect-test.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/sh
+
+# Usage: $0 /path/to/clusterclient-binary
+
+clientprog=${1:-./clusterclient}
+testname=moved-redirect-test
+
+# Sync processes waiting for CONT signals.
+perl -we 'use sigtrap "handler", sub{exit}, "CONT"; sleep 1; die "timeout"' &
+syncpid1=$!;
+perl -we 'use sigtrap "handler", sub{exit}, "CONT"; sleep 1; die "timeout"' &
+syncpid2=$!;
+
+# Start simulated redis node #1
+timeout 5s ./simulated-redis.pl -p 7403 -d --sigcont $syncpid1 <<'EOF' &
+EXPECT CONNECT
+EXPECT ["CLUSTER", "SLOTS"]
+SEND [[0, 16383, ["127.0.0.1", 7404, "nodeid7404"]]]
```

Comment:
This might be ok protocol wise, but I guess its less normal that the init-node is not part of the cluster.
Then later the key is moved here. But anyway, it should be handled

Reply:
The OOM corrections did not do it.
The async context got a pointer to the node in its `data` field, but the code after cluster_nodes_swap_ctx() deletes all nodes when swapping out the node information, the the async context tries to call the data-cleanup funktion `unlinkAsyncContextAndNode` but the node is already gone. Maybe there is a need to unregister the cleanup-func in the async context. Not sure how the idea is here

Reply:
Changed to a more realistic slot division: Move from the 1st to the 2nd instead of vice versa. The sanitizer warnings remain.